### PR TITLE
Rework tests

### PR DESCRIPTION
**Pull request by vsafonkin**
Test cases were moved to separated stages. Every stage contains clean up toolcache step before setup task and step to verify installed dotnet versions.

Step for verifying installed dotnet versions uses single powershell script (instead the separated ps and bash scripts for Windows and Linux/macOS platforms).

Issue #163

### CODE REVIEW

File: __tests__/clear-toolcache.ps1 [hunk 1]
```diff
@@ -0,0 +1,23 @@
+$os = $args[0]
+
+$linuxDotnetPaths = @("/usr/share/dotnet")
```
Comment (al-cheb) [inline __tests__/clear-toolcache.ps1:3]: nit: Why don't you use a hashtable?

```
$dotnetPaths = @{
    "Windows" = ...
}

$pathsToClear = $dotnetPaths[$os]
```

File: __tests__/clear-toolcache.ps1 [hunk 2]
```diff
@@ -0,0 +1,23 @@
+$os = $args[0]
```
Comment (al-cheb) [inline __tests__/clear-toolcache.ps1:1]: nit: Use block param. it's more obvious

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -2,18 +2,22 @@ name: Main workflow
...
+      - '**.md'
 
 jobs:
   build:
     runs-on: ${{ matrix.operating-system }}
     strategy:
       fail-fast: false
       matrix:
-        operating-system: [ubuntu-latest, windows-latest, macOS-latest]
+        operating-system: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04, windows-latest, macOS-latest]
```
Comment (AlenaSviridenko) [inline .github/workflows/workflow.yml:20]: why do you specify all 3 versions of Ubuntu but only the latest ones for Windows and macOS?

Comment (vsafonkin) [reply .github/workflows/workflow.yml:20]: We have only one macOS version on GitHub. Maybe it make sense to test `windows-2016` and `windows-2019` images.

File: __tests__/clear-toolcache.ps1 (hunk 1 above)
Comment (vsafonkin) [reply __tests__/clear-toolcache.ps1]: Good idea, thanks! Reworked.

File: .github/workflows/workflow.yml (hunk 3 above)
Comment (AlenaSviridenko) [reply .github/workflows/workflow.yml:20]: Actually, I think that there is no need to test on all Ubuntu images. You are verifying how different versions are resolved and then installed, right? If yes, then it shouldn't be different on different OS versions. 
At least, you are not specifying exact OS version in the new issue template, then it means it doesn't matter which exact version is used.

Comment (vsafonkin) [reply .github/workflows/workflow.yml:20]: I've changed os matrix similarly the other setup tasks, for example [setup-python](https://github.com/actions/setup-python/blob/main/.github/workflows/test-python.yml).

@maxim-lobanov , any thoughts?

Comment (maxim-lobanov) [reply .github/workflows/workflow.yml:20]: `macos-latest`, `ubuntu-latest`, `windows-latest` should be enough

File: .github/workflows/workflow.yml [hunk 4]
```diff
@@ -70,70 +58,86 @@ jobs:
...
+        uses: actions/checkout@v2
+      - name: Clear toolcache
+        shell: pwsh
+        run: __tests__/clear-toolcache.ps1 ${{ runner.os }}
+      - name: Setup dotnet 3.1.x
         uses: ./
         with:
-          dotnet-version: 2.0.x
+          dotnet-version: 3.1.x
+      - name: Setup dotnet 2.2.x
```
Comment (maxim-lobanov) [inline .github/workflows/workflow.yml:110]: May be we can use matrix to test different versions?

Comment (vsafonkin) [reply .github/workflows/workflow.yml:110]: The dotnet project that building in tests requires both 2.2 and 3.1 sdks, so, we have to install two versions side-by-side. As I see, we will be installed only one version if we use matrix, am I wrong?

**Review by AlenaSviridenko — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**